lib/translator/translation.cc

```diff
@@ -75,7 +77,16 @@ absl::Span<const nvme_defs::GenericQueueEntryCmd> Translation::GetNvmeCmds() {
 void Translation::AbortPipeline() {
   pipeline_status_ = StatusCode::kUninitialized;
   nvme_cmd_count_ = 0;
-  // TODO free allocated PRPs
+  FlushMemory();
+}
+
+void Translation::FlushMemory() {
+  for (int i = 0; i < kMaxCommandRatio; ++i) {
+    if (allocations_[i].data_addr != nullptr)
+      DeallocPages(allocations_[i].data_addr, allocations_[i].data_page_count);
+    if (allocations_[i].mdata_addr != nullptr)
+      DeallocPages(allocations_[i].mdata_addr, allocations_[i].mdata_page_count);
```

Comment:
We should set the addrs to `nullptr` afterwards so a reuse of the object doesn't cause double-frees. Similarly, `Translation` will now need an explicit constructor that sets the addrs to `nullptr` when the object is first created.